test/test_gpu_gather_ops.cpp

```diff
@@ -120,230 +128,250 @@ TEST_F(NVFuserTest, FusionScatter1DIndexZerosSelfTvSameShape_CUDA) {
 
 // Test the correctness of gather operator in different dimensions and selcted
 // dim.
-TEST_F(NVFuserTest, FusionTorchGatherAllRankAllSelectedDim_CUDA) {
+TEST_F(NVFuserTest, TorchGatherAllRankAllSelectedDim_CUDA) {
   const int max_dim_size = 64;
   std::srand(0);
   at::manual_seed(0);
   auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
   auto options_i = at::TensorOptions().dtype(at::kLong).device(at::kCUDA, 0);
-  for (int rank = 1; rank <= 5; ++rank) {
-    for (int dim = 0; dim < rank; ++dim) {
-      auto fusion_ptr = std::make_unique<Fusion>();
-      Fusion& fusion = *fusion_ptr.get();
-      FusionGuard fg(&fusion);
-
-      TensorView* tv1 = makeContigTensor(rank);
-      TensorView* tv_idx = makeContigTensor(rank, DataType::Int);
-      fusion.addInput(tv1);
-      fusion.addInput(tv_idx);
-      auto tv_out = torch_gather(tv1, dim, tv_idx);
-      fusion.addOutput(tv_out);
-
-      auto input_dims = randomVector(2, max_dim_size, rank);
-      auto index_dims = randomIndexVector(input_dims, 1, rank);
-      at::Tensor input = at::randn(input_dims, options);
-      at::Tensor input_idx =
-          at::randint(0, input_dims[dim], index_dims, options_i);
-      at::Tensor output = at::zeros(index_dims, options);
-
-      auto tv_out_ref = at::gather(input, dim, input_idx);
-      std::vector<c10::IValue> aten_inputs = {input, input_idx};
-
-      FusionExecutorCache executor_cache(std::move(fusion_ptr));
-      auto cg_outputs = executor_cache.runFusionWithInputs(aten_inputs);
-      testValidate(
-          &fusion, cg_outputs, aten_inputs, {tv_out_ref}, __LINE__, __FILE__);
+  for (const auto is_take_along : {false, true}) {
```

Comment:
Added `is_take_along` option so that both `torch.gather` and `numpy.take_along_axis` are tested. In the latter case, the index tensor has the same shape as the input tensor except for the indexed dimension